packages/core/src/checkout/checkout-service.ts

```diff
@@ -1432,6 +1432,17 @@ export default class CheckoutService {
         return state;
     }
 
+    /**
+     * Clear cache for a checkout extension when removing it from UI.
+     * This function should be used whenver an extension is removed from the UI.
+     *
+     * @alpha
+     * @param region - The name of an area where the extension should be presented.
+     */
+    clearExtensionCache(region: ExtensionRegion): void {
+        this._extensionMessenger.clearCacheByRegion(region);
+    }
```

Comment:
On `checkout-js` side, this will be called as part of an extension container component's `useEffect` return.
Therefore, it will run when an extension region is unmounted.